src/eigenlayer/ObolEigenLayerPodControllerFactory.sol

```diff
@@ -8,71 +9,65 @@ import {LibClone} from "solady/utils/LibClone.sol";
 /// @notice A factory contract for cheaply deploying ObolLidoEigenLayer.
 /// @dev The address returned should be used to as the EigenPod address
 contract ObolEigenLayerPodControllerFactory {
+  error Invalid_Owner();
+  error Invalid_WithdrawalAddress();
+  error Invalid_DelegationManager();
+  error Invalid_EigenPodManaager();
+  error Invalid_WithdrawalRouter();
+
+  using LibClone for address;
 
-    error Invalid_Owner();
-    error Invalid_Split();
-    error Invalid_DelegationManager();
-    error Invalid_EigenPodManaager();
-    error Invalid_WithdrawalRouter();
+  event CreatePodController(address indexed controller, address indexed withdrawalAddress, address owner);
 
-    using LibClone for address;
+  ObolEigenLayerPodController public immutable controllerImplementation;
 
-    event CreatePodController(
-        address indexed controller,
-        address indexed split,
-        address owner
-    );
-    
-    ObolEigenLayerPodController public immutable controllerImplementation;
+  constructor(
+    address feeRecipient,
+    uint256 feeShare,
+    address delegationManager,
+    address eigenPodManager,
+    address withdrawalRouter
+  ) {
+    if (delegationManager == address(0)) revert Invalid_DelegationManager();
+    if (eigenPodManager == address(0)) revert Invalid_EigenPodManaager();
+    if (withdrawalRouter == address(0)) revert Invalid_WithdrawalRouter();
 
-    constructor(
-        address feeRecipient,
-        uint256 feeShare,
-        address delegationManager,
-        address eigenPodManager,
-        address withdrawalRouter
-    ) {
-        if (delegationManager == address(0)) revert Invalid_DelegationManager();
-        if (eigenPodManager == address(0)) revert Invalid_EigenPodManaager();
-        if (withdrawalRouter == address(0)) revert Invalid_WithdrawalRouter();
+    controllerImplementation =
+      new ObolEigenLayerPodController(feeRecipient, feeShare, delegationManager, eigenPodManager, withdrawalRouter);
+    // initialize implementation
+    controllerImplementation.initialize(feeRecipient, feeRecipient);
+  }
 
-        controllerImplementation = new ObolEigenLayerPodController(
-            feeRecipient,
-            feeShare,
-            delegationManager,
-            eigenPodManager,
-            withdrawalRouter
-        );
+  /// Creates a minimal proxy clone of implementation
+  /// @param owner address of owner
+  /// @param withdrawalAddress address of withdrawalAddress
```

Comment:
Just for my understanding, we're separating out the ability to control the controller contract to `owner`, and then setting another address for the funds to flow to (`withdrawalAddress`), which in practice could be an OWR or a SplitProxy? Out of interest, in a standard eigenpodcontroller, there is no distinction between the controller and recipient of funds, and this is functionality we are adding, right?

Reply:
yes, that's correct